configs/Default (Linux).sublime-keymap

```diff
@@ -0,0 +1,742 @@
+[
```

Comment:
Again, my goal, which I know is not compatible with xi-terms goal is to have compatibility with sublime. Thus, I used their keymap-files, so existing user-configs could just be copied in.

Reply:
> which I know is not compatible with xi-terms goal is to have compatibility with sublime

Oh I had misunderstood that part. I thought you wanted to make something somewhat similar to Sublime, but not 100% compatible. In that case, would you mind if I took whatever I'm interested in from your changes, and adapt it to what I envision for xi-term?

Reply:
Yes, absolutely. That was my suggestion from the beginning :-)
To be clear: I don't have 100% compatibility as a hard requirement, although it would be nice and handy. 
But since you are, as far as I understand, going for vim-like modular editing my editor (be it 100% sublime compatible or not) will definitely not be compatible with xi-term. 
So i figured a hard fork is my only path. 
It certainly would be nice join forces, though. But currently I don't really see how.

Reply:
> But since you are, as far as I understand, going for vim-like modular editing my editor

Oh I don't have a clear path actually. I don't really want to make a vim clone for multiple reasons (I don't like VimScript in the first place). It's true that I'd like to provide a way to do vim-like editing because that's what I'm used to, but I'd like to make that optional, and allow users to use their own keybindings. Emacs is doing that with evil mode for instance (which is what I use), and I really like that approach.

> It certainly would be nice join forces, though. But currently I don't really see how.

As far as I can tell the only thing that is _really_ specific to Sublime in this PR is the configuration format and the default bindings. To me, the Sublime bindings are fine, and it's easy to use different ones anyway. The only thing I'm not fond is using json for the config file but I think I can live with that for now and switch to something else in the future (there are solutions if we want to keep supporting json: having multiple parsers, or providing a tool to convert config files, etc.).

I've been using vim for a long time, but I'm not opposed to ideas from other editors because I'm not convinced that making re-implementing an existing editor is likely to succeed. First, it's hard to replicate the exact behavior of another editor. Second, there are behaviors don't want to keep (I don't like vim's registers I prefer emacs kill ring, and I don't like emacs undo system, I prefer vim's for instance). Finally, users won't switch to something new if it's exactly the same thing.

So yeah, I'm open so non-vim-like behavior :)

Reply:
> Finally, users won't switch to something new if it's exactly the same thing.

Absolutely. The only reason I started this was because there simply is no terminal editor like sublime at all. There is one without multiple cursors and such, but that is abandoned (and written in JS. Of course...).

> So yeah, I'm open to non-vim-like behavior :)

Ok, cool! So maybe we could join forces after all. That would be nice. 

Yes the only really sublime-specific stuff is the json-format (including structs for parsing it). 
Once we have the translated map of Temion-`Keyevents` and `Command`, it doesn't matter anymore. 

And since the Command-structs for parsing the json could get different names with the serde-macros, we can hide sublime heritage there as well. 

Only, in the prompt, I currently expose sublimes names to the user as well. There could of course be another layer in there as well. Tried to keep it simple for now.

The only blocker I see would be the modal editing (insert mode, vs. command mode), if you envision that for your editor.

Reply:
> So maybe we could join forces after all.

I hope so!

> Once we have the translated map of Temion-Keyevents and Command, it doesn't matter anymore.

Yeah, that's why I'm not to worried about using JSON for now. It should be easy to change in the future.

> The only blocker I see would be the modal editing (insert mode, vs. command mode), if you envision that for your editor.

Well for now no frontend has it, and it's unclear whether that requires support in xi-core or not. Plus there are tons of other features to add first, so it's not a priority for me right now.

Reply:
> Well for now no frontend has it, and it's unclear whether that requires support in xi-core or not. Plus there are tons of other features to add first, so it's not a priority for me right now.

Alright. Although I see (for the base functionality) no reason to involve xi-core. If editor.rs has a mode it can interpret a keypress-event like "w" as either `Insert(w)` or `RelativeMove{by: word}` and act accordingly.